### PR TITLE
[BatchConfirmer] Apply state transitions for full batches

### PR DESCRIPTION
**Pull request by ian-shim**
## Why are these changes needed?
This PR makes `BatchConfirmer` transition full batch state to `Failed` or `Attested`. 
It also improves the test coverage for batch confirmer. 

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/batcher/batch_confirmer.go [hunk 1]
```diff
@@ -170,15 +174,7 @@ func (b *BatchConfirmer) updateConfirmationInfo(
...
 			// generate inclusion proof
-			blobHeader := batchData.blobHeaders[blobIndex]
-
-			blobHeaderHash, err := blobHeader.GetBlobHeaderHash()
-			if err != nil {
-				b.logger.Error("failed to get blob header hash", "err", err)
-				blobsToRetry = append(blobsToRetry, batchData.blobs[blobIndex])
-				continue
-			}
-			merkleProof, err := batchData.merkleTree.GenerateProof(blobHeaderHash[:], 0)
```
Comment (ian-shim) [inline disperser/batcher/batch_confirmer.go:181]: Proof should be generated by index

File: disperser/batcher/batch_confirmer.go [hunk 2]
```diff
@@ -220,6 +217,11 @@ func (b *BatchConfirmer) updateConfirmationInfo(
 		if updateConfirmationInfoErr != nil {
 			b.logger.Error("error updating blob confirmed metadata", "err", updateConfirmationInfoErr)
 			blobsToRetry = append(blobsToRetry, batchData.blobs[blobIndex])
+		} else {
+			err = b.MinibatchStore.UpdateBatchStatus(ctx, batchData.batchID, BatchStatusAttested)
```
Comment (mooselumph) [inline disperser/batcher/batch_confirmer.go:221]: I think this should be outside of the for-loop, right?

**Review by mooselumph — COMMENTED** [review]
Just one concern

File: disperser/batcher/batch_confirmer.go (hunk 2 above)
Comment (ian-shim) [reply disperser/batcher/batch_confirmer.go:221]: Thanks for catching this

**Review by pschork — APPROVED** [review]

**Pull request merged**